services/galley/src/Galley/API/Update.hs

```diff
@@ -1465,8 +1465,7 @@ addBot lusr zcon b = do
     teamConvChecks :: ConvId -> TeamId -> Sem r ()
     teamConvChecks cid tid = do
       tcv <- E.getTeamConversation tid cid
-      when (maybe True (view managedConversation) tcv) $
-        throw NoAddToManaged
+      when (isNothing tcv) $ throw NoAddToManaged
```

Comment:
@pcapriotti , does this check make sense at all? We're throwing out managed conversations with this PR, yet here we say that a user or bot cannot be added to a managed conversation directly. Any objection to gettting rid of this check as well?

Reply:
Yes, this throw is unreachable, since at this point we already know that it is a team conversation (look at the call site).
Please remove this whole function and the `NoAddToManaged` constructor.